src/Knp/Snappy/AbstractGenerator.php

```diff
@@ -493,6 +493,22 @@ protected function buildCommand($binary, $input, $output, array $options = [])
         return $command;
     }
 
+    /**
+     * Handles non-locale-aware floats.
+     *
+     * @param mixed $value
+     *
+     * @return mixed
+     */
+    private function handleCommandFloat($value)
```

Comment:
I think it's better to handle float option values for `unitreal` and `float` options only (see https://wkhtmltopdf.org/usage/wkhtmltopdf.txt for the complete list of options and their types).

What about refactoring the code this way?
```php
private function buildCommandOptions(array $options)
{
    // Move format option logic here
    // https://github.com/KnpLabs/snappy/blob/fd6fa25fcab3b789ef02847e95eedb8170cc66e8/src/Knp/Snappy/AbstractGenerator.php#L453-L481
}

private function escapeCommandOptionValue(string $key, $value)
{
    if (in_array($key, ['margin-bottom', 'margin-top', '...'])) {
        if (is_float($value)) {
            return trim(sprintf('%F', $value), '0');
        }
    }

    // Other special cases...

    return escapeshellarg($value);
}
```

What do you think?